doc/notes/189_reporting_ant.adoc

```diff
@@ -0,0 +1,224 @@
+
+The amount of time elapsed since the Protocol Verifier was started shall
+be displayed.
+
+The units shall be in time duration format.
+
+===== 3.1.2.2 Protocol Verifier Worker Count
+
+The Protocol Verifier scales up and down to handle various audit event
+stream loads efficiently.  An integer number representing the current
+number of Protocol Verifier "worker" instances shall be shown.  A max
+count and a minimum count shall display the maximum and minimum number
+respectively over a period of time (such as 24 hours, or since the
+Protocol Verifier started).
+
+There are no units to this measurement.
+
+===== 3.1.2.3 Count of Assigned Jobs (Jobs in Progress)
+
+The number of jobs which Job Management has assigned to a worker shall be
+displayed.
+
+There are no units to this measurement.
+
+===== 3.1.2.4 Count of Unassigned Jobs
+
+The number of jobs which Job Management has seen but has not assigned to a
+work shall be displayed.
+
+There are no units to this measurement.
+
+=== 3.2 Other Requirements
+
+* Reporting shall be supplied through a Kafka message broker topic.
+* The message shall be consumable by <<dr-3, Apache NiFi>>.
+
+== 4 Analysis
+
+Analysis for Protocol Verifier Reporting has been performed by the Munin
+engineering team with little outside input.  However, as understanding of
+protocol verification grew, an understanding of what would be useful on
+the back end grew.  The above dashboard and reporting requirements emerged
+from weeks of running the Protocol Verifier with simulated input audit
+event streams.
+
+=== 4.1 Questions
+
+While testing the Protocol Verifier several questions arose.
+
+* How fast is the PV running?
+* How do we know when a job has failed?
+* How do we identify a failed job?
+* Which job types fail most often?
+* How many jobs have failed?
+* How many jobs have succeeded?
+* How many jobs are receiving alarm conditions?
+* What are the success/failure/alarm percentages?
+* What are the names of the jobs that receive alarm conditions?
+* Is the PV running optimally?
+* How long has the PV been running?
+* How does the throughput change when I adjust timing parameters?
+
+It is expected that more questions and reporting requirements will emerge.
+But for now, even basic usage of the Protocol Verifier raises these
+questions.  Thus, let us build reporting to answer the questions we have
+uncovered so far.
+
+== 5 Work Required
+
+=== 5.1 Protocol Verifier Instrumentation
+
+. Craft a mock-up dashboard using a graphics drawing tool (e.g.  PowerPoint).
++
+.Protocol Verifier Dashboard (MOCK-UP)
+image::dashboard1.png[Protocol Verifier Dashboard MOCK-UP]
+
+. Supplied with the requirements from this document, analyse the logging and
+  reporting instrumentation supplied by the Protocol Verifier.
+. Update the PV with sufficient logging and reporting to generate the data
+  needed by the reporting dashboard.
+
+=== 5.2 Protocol Verifier Back End
+
+* Build a dashboard using appropriate technology.
```

Comment:
As per my question on line 22 above - in my mind I had thought that the client would be consuming the reported information in a dashboard that they were either responsible for building, or consuming in extensions to existing dashboard(s) - do I need to reset my thinking?